tests/retrieval/helpers.py

```diff
@@ -30,10 +30,40 @@
 from tests.retrieval.inputs import _input_retrieval_scores_mismatching_sizes_func as _irs_mis_sz_fn
 from tests.retrieval.inputs import _input_retrieval_scores_no_target as _irs_no_tgt
 from tests.retrieval.inputs import _input_retrieval_scores_wrong_targets as _irs_bad_tgt
-from torchmetrics.utilities.data import get_group_indexes
 
 seed_all(42)
 
+# a version of get_group_indexes that depends on NumPy is here to avoid this dependency for the full library
+
+
+def get_group_indexes(indexes: Union[Tensor, np.ndarray]) -> List[Union[Tensor, np.ndarray]]:
```

Comment:
Here you still use np

Reply:
This is in tests, we do use scikit-learn in tests as well, which is fine, I'm only worried about the core library being lean.